compiler/src/dmd/expressionsem.d

```diff
@@ -12879,6 +12879,14 @@ Expression binSemantic(BinExp e, Scope* sc)
     if (e2x.op == EXP.type)
         e2x = resolveAliasThis(sc, e2x);
 
+    if (e1x.op == EXP.type || e2x.op == EXP.type)
+    {
+        if (e1x.op == EXP.type)
+            e1x.error("type `%s` is not an expression", e1x.toChars());
+        else
+            e2x.error("type `%s` is not an expression", e2x.toChars());
+        return ErrorExp.get();
+    }
```

Comment:
Call `if (e1x.checkType() || e2x.checkType())` - unless there is something I am not understanding between this addition and the method.

Reply:
Changed to use `checkValue`. I tried `checkType` but that was erroring when calling a template function.